Bugix/admin panel fixes

neuromanzer opened the pull request.

neuromanzer committed:
34-кабинет-админа
убран вывод пароля при редактировании(html&js)
поправлено редактирование юзера(модалка)

neuromanzer committed:
34-кабинет-админа
в модалке редактирования юзера исправлено отображение ролей

neuromanzer committed:
34-кабинет-админа
при созданнии нового юзера ввод невалидного имейла приводит к сообщению об ошибке

neuromanzer committed:
34-кабинет-админа
поправлены стили кнопок редактирования и удаления

neuromanzer committed:
34-кабинет-админа
поправлены стили кнопок редактирования и удаления #2

neuromanzer committed:
34-кабинет-админа
рабочий коммит

neuromanzer committed:
34-кабинет-админа
рабочий коммит

neuromanzer committed:
Merge branch 'dev' into bugix/admin-panel-fixes

# Conflicts:
#	src/main/java/com/jm/online_store/model/User.java
#	src/main/java/com/jm/online_store/service/UserServiceImpl.java
#	src/main/resources/application.properties

tediik commented on docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+version: '3.8'
+services:
+  db:
+    image: mysql:latest
+    restart: always
+    environment:
+      MYSQL_DATABASE: 'online_store'
+      # MYSQL_USER: 'db_user'
+      # MYSQL_PASSWORD: 'db_password'
+      MYSQL_ROOT_PASSWORD: 'root'
+    ports:
+      - '3306:3306'
+    expose:
+      - '3306'
+    volumes:
+      - db_vol:/var/lib/mysql
+volumes:
+  db_vol:
```
Зачем нам лишние конфиги в проекте? Настрой у себя gitignore, файл убирай из коммита

tediik commented on src/main/java/com/jm/online_store/model/User.java:
```diff
@@ -60,7 +63,7 @@
 
     @ManyToMany(fetch = FetchType.EAGER,
-            cascade = CascadeType.REFRESH)
+            cascade = CascadeType.ALL)
```
Нам точно нужен будет ALL?

tediik commented on src/main/java/com/jm/online_store/service/UserServiceImpl.java:
```diff
@@ -109,14 +138,17 @@ public void regNewAccount(User userForm) {
         }
 
-        Set<Role> userSetRoles = Collections.singleton(roleRepository.findByName("ROLE_CUSTOMER").get());
+        Role userRole = roleRepository.findByName("ROLE_CUSTOMER").get();
+        Set<Role> userSetRoles = new HashSet<>();
+        userSetRoles.add(userRole);
```
Чем не устроил старый вариант?

tediik commented on src/main/resources/static/main.js:
```diff
@@ -121,14 +125,17 @@ $(document).ready(function () {
+                    console.log(jqXhr);
+                    console.log(textStatus);
                     console.log(errorThrown);
+                    alert(jqXhr.responseJSON.message);
```
Хм.. не уверен, что нам нужны Алерты, давай уберем

tediik reviewed (changes requested):
Ты бросаешь новые excepton но я правильно понимаю, что нигде это не обрабатываешь??

tediik commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Кабинет админа](https://trello.com/c/XBN5bNP4/34-%D0%BA%D0%B0%D0%B1%D0%B8%D0%BD%D0%B5%D1%82-%D0%B0%D0%B4%D0%BC%D0%B8%D0%BD%D0%B0)

neuromanzer committed:
34-кабинет-админа
убраны логи в js

neuromanzer committed:
34-кабинет-админа
fetchType на роли REFRESH

neuromanzer committed:
34-кабинет-админа
убран конфиг докера

neuromanzer committed:
34-кабинет-админа
добавлены алерты bootstrap на странице добавления юзера

neuromanzer committed:
34-кабинет-админа
добавлены алерты на добавление и редактирование юзера + рефакторинг метода UserServiceImpl#activateUser

neuromanzer committed:
Merge branch 'dev' into bugix/admin-panel-fixes

# Conflicts:
#	src/main/java/com/jm/online_store/controller/rest/UserRestController.java
#	src/main/java/com/jm/online_store/model/User.java
#	src/main/java/com/jm/online_store/service/impl/UserServiceImpl.java

tediik commented on src/main/java/com/jm/online_store/config/security/SecurityConfig.java:
```diff
@@ -61,6 +62,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 // делаем страницу регистрации недоступной для авторизированных пользователей
                 .authorizeRequests()
                 .antMatchers("/", "/login", "/news/**", "/registration", "/css/**").permitAll()
+                .antMatchers("/test-reg").permitAll()
```
это что за url?

tediik commented on src/main/java/com/jm/online_store/config/security/SecurityConfig.java:
```diff
@@ -33,7 +33,8 @@ public void configure(AuthenticationManagerBuilder auth) throws Exception {
     protected void configure(HttpSecurity http) throws Exception {
-        http.oauth2Login().loginPage("/login").userInfoEndpoint().userService(facebookOAuth2UserService).and().and().authorizeRequests().antMatchers("/customer").hasRole("CUSTOMER");
+        http.oauth2Login().loginPage("/login").userInfoEndpoint().userService(facebookOAuth2UserService)
+                .and().and().authorizeRequests().antMatchers("/customer").hasRole("CUSTOMER");
```
мне кажется 1 .and() лишний

neuromanzer replied:
изначально было также 2 .and
при попытке убрать один выделяется красным .authorizeRequests()

tediik commented on src/main/java/com/jm/online_store/controller/simple/TestRegController.java:
```diff
@@ -0,0 +1,44 @@
+        model.addAttribute("message", "User added successfully!");
+        return "successfulRegister";
+    }
+}
```
Надо бы пояснить, для чего нам этот контроллер?

tediik commented on src/main/resources/application.properties:
```diff
@@ -1,6 +1,6 @@
 # Database/JPA configs
 spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
-spring.datasource.url=jdbc:mysql://localhost:3306/online_store?useUnicode=yes&characterEncoding=UTF-8&useSSL=false&serverTimezone=Europe/Moscow
+spring.datasource.url=jdbc:mysql://localhost:3306/online_store?useUnicode=yes&characterEncoding=UTF-8&allowPublicKeyRetrieval=true&useSSL=false&serverTimezone=UTC
```
убери, верни старую строчку

tediik commented on src/main/resources/templates/test-reg.html:
```diff
@@ -0,0 +1,52 @@
+<script src="https://code.jquery.com/jquery-3.5.1.slim.min.js" integrity="sha384-DfXdz2htPH0lsSSs5nCTpuj/zy4C+OGpamoFVy38MVBnE+IbbVYUew+OrCXaRkfj" crossorigin="anonymous"></script>
+<script src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js" integrity="sha384-Q6E9RHvbIyZFJoft+2mJbHaEWldlvI9IOYy5n3zV9zzTtmI3UksdQRVvoxMfooAo" crossorigin="anonymous"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.0/js/bootstrap.min.js" integrity="sha384-OgVRvuATP1z7JjHLkuOU7Xw704+h835Lr+6QL9UvYjZE3Ipu6Tp75j7Bh/kR0JKI" crossorigin="anonymous"></script>
+</body>
```
Нужно пояснение что это за страничка?

neuromanzer committed:
34-кабинет-админа
убран test-reg контроллер и html страница
в проперти прежняя настройка spring.datasource.url
